src/main/scala/org/codeoverflow/chatoverflow/requirement/service/rcon/RconConnector.scala

```diff
@@ -0,0 +1,129 @@
+package org.codeoverflow.chatoverflow.requirement.service.rcon
+
+import java.io.{DataInputStream, InputStream, OutputStream}
+import java.net.{Socket, SocketException}
+import java.nio.{ByteBuffer, ByteOrder}
+import java.util.Random
+
+import org.codeoverflow.chatoverflow.WithLogger
+import org.codeoverflow.chatoverflow.connector.Connector
+
+class RconConnector(override val sourceIdentifier: String) extends Connector(sourceIdentifier) with WithLogger {
+  override protected var requiredCredentialKeys: List[String] = List("password", "address")
+  override protected var optionalCredentialKeys: List[String] = List("port")
+
+  private var socket: Socket = _
+  private var outputStream: OutputStream = _
+  private var inputStream: InputStream = _
+  private var requestId: Int = 0
+  private var loggedIn = false
+
+  def sendCommand(command: String): String = {
+    if (!loggedIn) {
+      logger error "Could not execute RCON Command due to wrong password or no connection"
+      return null
+    }
+    logger debug s"Sending $command to RCON"
+    requestId += 1
+    if (write(2, command.getBytes("ASCII"))) {
+      return read()
+    }
+    null
+  }
+
+
+  /**
+    * Starts the connector, e.g. creates a connection with its platform.
+    */
+  override def start(): Boolean = {
+    logger info s"Starting rcon connection to ${credentials.get.getValue("address").get}"
+    var port: Int = 25575
+    if (credentials.get.exists("port")) {
+      port = credentials.get.getValue("port").get.toInt
+      if (port < 1 || port > 65535) {
+        return false
+      }
+    }
+    socket = new Socket(credentials.get.getValue("address").get, port)
+    socket.setKeepAlive(true)
+    outputStream = socket.getOutputStream
+    inputStream = socket.getInputStream
+    login()
+    Thread.sleep(5000)
+    true
+  }
+
+  private def login(): Unit = {
+    requestId = new Random().nextInt(Integer.MAX_VALUE)
+    logger info "Logging RCON in..."
+    val password = credentials.get.getValue("password").get
+    if (write(3, password.getBytes("ASCII"))) {
+      if (read() == null) {
+        logger error "Could not log in to RCON Server. Password is Wrong!"
+      } else {
+        logger debug "Login to RCON was successful"
+        loggedIn = true
+      }
+    }
+  }
```

Comment:
Instead of setting logged in it would be better to just return if the login was successfull and if not close the socket and return false in the `start()` function.   

That way the connector isn't enabled if login fails.  
All other connectors work that way, just have a look at them.